sdk/storage/Azure.Storage.Blobs/tests/ContainerClientTests.cs

```diff
@@ -1541,6 +1556,47 @@ public async Task DeleteBlobAsync()
             }
         }
 
+        #region Secondary Storage
+        [Test]
+        public async Task ListContainersSegmentAsync_SecondaryStorageFirstRetrySuccessful()
+        {
+            var testExceptionPolicy = await this.PerformSecondaryStorageTest(1); // one GET failure means the GET request should end up using the SECONDARY host
+            this.AssertSecondaryStorageFirstRetrySuccessful(this.SecondaryStorageTenantPrimaryHost(), SecondaryStorageTenantSecondaryHost(), testExceptionPolicy);
+        }
+
+
+        private async Task<object> GetProperties(object client)
+        {
+            BlobContainerClient blobContainerClient = client as BlobContainerClient;
+            var properties = await blobContainerClient.GetPropertiesAsync();
+            return properties;
+        }
+
+        private bool HasProperties(object response)
+        {
+            Response<ContainerItem> properties = (Response<ContainerItem>) response;
+            return properties.GetRawResponse().Status != 404;
+        }
+
+        private async Task<TestExceptionPolicy> PerformSecondaryStorageTest(int numberOfReadFailuresToSimulate, bool retryOn404 = false)
```

Comment:
Nit - only one method ever calls this?

Reply:
Eh, it matches the pattern from the other classes and would make it easier to add additional tests in the future.